[5.4.0] Typescript 4.4.4 (latest) not supported anymore

<!--
Please don't ignore this template.

If you ignore it, we're just going to respond asking you to fill it out, which wastes everyone's time.
The more relevant information you can include, the faster we can find the issue and fix it without asking you for more info.
-->

<!--
🚨 STOP 🚨 𝗦𝗧𝗢𝗣 🚨 𝑺𝑻𝑶𝑷 🚨

This issue template is only for problems specifically with the `@typescript-eslint/typescript-estree` package.

If you have a problem with a specific lint rule, please back out and select the `@typescript-eslint/eslint-plugin` template.
-->

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/TROUBLESHOOTING.md) and my problem is not listed.

**Repro**

Using Typescript 4.4.4 (latest non-rc release)

**Expected Result**

linting should pass

**Actual Result**

```
node_modules/@typescript-eslint/typescript-estree/dist/ts-estree/estree-to-ts-node-types.d.ts:40:42 - error TS2694: Namespace 'ts' has no exported member 'AssertEntry'.
```

![image](https://user-images.githubusercontent.com/3777724/141995056-82218311-93e4-4320-825c-9cdf9dc6d4bd.png)

**Additional Info**

See above

**Versions**

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/typescript-estree` | `5.4.0` |
| `TypeScript`                           | `4.4.4` |
| `node`                                 | `14.17.6` |

--- a/packages/typescript-estree/src/ts-estree/ts-nodes.ts
+++ b/packages/typescript-estree/src/ts-estree/ts-nodes.ts
@@ -8,6 +8,8 @@ declare module 'typescript' {
   export interface TemplateLiteralTypeNode extends ts.Node {}
   export interface PrivateIdentifier extends ts.Node {}
   export interface ClassStaticBlockDeclaration extends ts.Node {}
+  export interface AssertClause extends ts.Node {}
+  export interface AssertEntry extends ts.Node {}
   /* eslint-enable @typescript-eslint/no-empty-interface */
 }
 

